core/market/decentralized/src/testing/events_helper.rs

```diff
@@ -30,6 +45,18 @@ pub mod provider {
             _ => anyhow::bail!("Invalid event Type. ProposalEvent expected"),
         })
     }
+
+    pub async fn query_proposal(
+        market: &Arc<MarketService>,
+        offer_id: &SubscriptionId,
+        i: u8,
```

Comment:
the name of this arg is my fault, but I'm surprised that you've not changed it ;)